src/operator/image/center_crop-inl.h

```diff
@@ -0,0 +1,169 @@
+    dst_h = src_h;
+  }
+  if (src_w < dst_w) {
+    dst_h = static_cast<int>((dst_h * src_w) / dst_w);
+    dst_w = src_w;
+  }
+  return SizeParam(dst_h, dst_w);
+}
+
+bool CenterCropShape(const nnvm::NodeAttrs& attrs,
+                             std::vector<TShape> *in_attrs,
+                             std::vector<TShape> *out_attrs) {
+  // input attrs should only be (h, w, c) or (n, h, w, c)
+  CHECK((in_attrs->at(0).ndim() == 3U) || (in_attrs->at(0).ndim() == 4U))
+    << "Input image dimension should be 3 or 4 but got "
+    << in_attrs->at(0).ndim();
+  const auto& ishape = (*in_attrs)[0];
+  const CenterCropParam& param = nnvm::get<CenterCropParam>(attrs.parsed);
+  const auto size = GetHeightAndWidthFromSize(param);
+
+  CHECK((size.height > 0) && (size.width > 0))
+      << "Input height and width must be greater than 0";
+  if (ishape.ndim() == 3) {
+    SHAPE_ASSIGN_CHECK(*out_attrs, 0, TShape({size.height, size.width, ishape[C]}));
+  } else {
+    SHAPE_ASSIGN_CHECK(*out_attrs, 0, TShape({ishape[N], size.height, size.width, ishape[kC]}));
+  }
+
+  return true;
+}
+
+void CenterCrop(const nnvm::NodeAttrs &attrs,
+                   const OpContext &ctx,
+                   const std::vector<TBlob> &inputs,
+                   const std::vector<OpReqType> &req,
+                   const std::vector<TBlob> &outputs) {
+  CHECK_EQ(outputs.size(), 1U);
+  CHECK((inputs[0].ndim() == 3) || (inputs[0].ndim() == 4))
+      << "Input data must be (h, w, c) or (n, h, w, c)";
+  const CenterCropParam& param = nnvm::get<CenterCropParam>(attrs.parsed);
+  const auto size = GetHeightAndWidthFromSize(param);
+  auto need_resize = false;
+  int h, w;
+  if (inputs[0].ndim() == 3) {
+    h = inputs[0].shape_[0];
+    w = inputs[0].shape_[1];
+  } else {
+    h = inputs[0].shape_[1];
+    w = inputs[0].shape_[2];
+  }
+  const auto new_size = ScaleDown(SizeParam(h, w), size);
+  if ((new_size.height != size.height) || (new_size.width != size.width)) {
+    need_resize = true;
+  }
+  const auto x0 = static_cast<int>((w - new_size.width) / 2);
+  const auto y0 = static_cast<int>((h - new_size.height) / 2);
+  if (inputs[0].ndim() == 3) {
+    if (need_resize) {
+      CropImpl(x0, y0, new_size.height, new_size.width, inputs, outputs, size, param.interp);
+    } else {
+      CropImpl(x0, y0, new_size.height, new_size.width, inputs, outputs);
+    }
+  } else {
+    const auto batch_size = inputs[0].shape_[0];
+    const auto input_offset = inputs[0].shape_[kH] * inputs[0].shape_[kW] * inputs[0].shape_[kC];
+    int output_offset;
+    if (need_resize) {
+      output_offset = size.height * size.width * outputs[0].shape_[kC];
+    } else {
+      output_offset = new_size.height * new_size.width * outputs[0].shape_[kC];
+    }
+    #pragma omp parallel for
+    for (auto i = 0; i < batch_size; ++i) {
+      if (need_resize) {
+        CropImpl(x0, y0, new_size.height, new_size.width, inputs, outputs,
```

Comment:
are these calls thread safe. good way to test would be to set omp_num_threads to 1 and test and compare with omp_num_threads as default.